### PR TITLE
NCBI datasets container

### PR DESCRIPTION
**Pull request by erinyoung**
I have an evil plan to create this dockerfile to make downloading files from NCBI for tests easier.

This container includes NCBI's datasets and dataformat.

Pull Request (PR) checklist:
- [X] Include a description of what is in this pull request in this message.
- [X] The dockerfile successfully builds to a test target for the user creating the PR. (i.e. `docker build --tag samtools:1.15test --target test docker-builds/samtools/1.15` )
- [X] Directory structure as name of the tool in lower case with special characters removed with a subdirectory of the version number (i.e. `spades/3.12.0/Dockerfile`)
   - [X] (optional) All test files are located in same directory as the Dockerfile (i.e. `shigatyper/2.0.1/test.sh`)
- [X] Create a simple container-specific [README.md](https://github.com/StaPH-B/docker-builds/blob/master/.github/workflow-templates/readme-template.md) in the same directory as the Dockerfile (i.e. `spades/3.12.0/README.md`)
   - [X] If this README is longer than 30 lines, there is an explanation as to why more detail was needed
- [X] Dockerfile includes the recommended [LABELS](https://github.com/StaPH-B/docker-builds/blob/master/dockerfile-template/Dockerfile#L8-L18) 
- [X] Main [README.md](https://github.com/StaPH-B/docker-builds/blob/master/README.md) has been updated to include the tool and/or version of the dockerfile(s) in this PR
- [X] [Program_Licenses.md](https://github.com/StaPH-B/docker-builds/blob/master/Program_Licenses.md) contains the tool(s) used in this PR and has been updated for any missing

### CODE REVIEW

**Comment by erinyoung** [comment]
Finishing my testing for the day. 

I'm currently getting stuck at 
```
#6 [app 2/5] RUN apt-get update && apt-get install -y --no-install-recommends  wget  unzip  ca-certificates  apt-get autoclean && rm -rf /var/lib/apt/lists/*
#6 0.622 Get:1 http://security.ubuntu.com/ubuntu jammy-security InRelease [110 kB]
#6 0.763 Get:2 http://archive.ubuntu.com/ubuntu jammy InRelease [270 kB]
#6 0.916 Get:3 http://security.ubuntu.com/ubuntu jammy-security/universe amd64 Packages [128 kB]
#6 1.021 Get:4 http://security.ubuntu.com/ubuntu jammy-security/multiverse amd64 Packages [4644 B]
#6 1.022 Get:5 http://security.ubuntu.com/ubuntu jammy-security/restricted amd64 Packages [276 kB]
#6 1.131 Get:6 http://security.ubuntu.com/ubuntu jammy-security/main amd64 Packages [305 kB]
#6 2.703 Get:7 http://archive.ubuntu.com/ubuntu jammy-updates InRelease [114 kB]
#6 3.211 Get:8 http://archive.ubuntu.com/ubuntu jammy-backports InRelease [99.8 kB]
#6 3.488 Get:9 http://archive.ubuntu.com/ubuntu jammy/restricted amd64 Packages [164 kB]
#6 3.722 Get:10 http://archive.ubuntu.com/ubuntu jammy/main amd64 Packages [1792 kB]
#6 11.17 Get:11 http://archive.ubuntu.com/ubuntu jammy/multiverse amd64 Packages [266 kB]
#6 12.95 Get:12 http://archive.ubuntu.com/ubuntu jammy/universe amd64 Packages [17.5 MB]
#6 122.4 Get:13 http://archive.ubuntu.com/ubuntu jammy-updates/restricted amd64 Packages [319 kB]
#6 124.5 Get:14 http://archive.ubuntu.com/ubuntu jammy-updates/universe amd64 Packages [248 kB]
#6 126.1 Get:15 http://archive.ubuntu.com/ubuntu jammy-updates/multiverse amd64 Packages [7791 B]
#6 126.2 Get:16 http://archive.ubuntu.com/ubuntu jammy-updates/main amd64 Packages [602 kB]
#6 130.3 Get:17 http://archive.ubuntu.com/ubuntu jammy-backports/universe amd64 Packages [5814 B]
#6 130.3 Fetched 22.2 MB in 2min 10s (171 kB/s)
#6 130.3 Reading package lists...
#6 131.3 Reading package lists...
#6 132.1 Building dependency tree...
#6 132.3 Reading state information...
#6 132.3 E: Unable to locate package apt-get
#6 132.3 E: Unable to locate package autoclean
#6 ERROR: process "/bin/sh -c apt-get update && apt-get install -y --no-install-recommends  wget  unzip  ca-certificates  apt-get autoclean && rm -rf /var/lib/apt/lists/*" did not complete successfully: exit code: 100
------
 > [app 2/5] RUN apt-get update && apt-get install -y --no-install-recommends  wget  unzip  ca-certificates  apt-get autoclean && rm -rf /var/lib/apt/lists/*:
#6 12.95 Get:12 http://archive.ubuntu.com/ubuntu jammy/universe amd64 Packages [17.5 MB]
#6 122.4 Get:13 http://archive.ubuntu.com/ubuntu jammy-updates/restricted amd64 Packages [319 kB]
#6 124.5 Get:14 http://archive.ubuntu.com/ubuntu jammy-updates/universe amd64 Packages [248 kB]
#6 126.1 Get:15 http://archive.ubuntu.com/ubuntu jammy-updates/multiverse amd64 Packages [7791 B]
#6 126.2 Get:16 http://archive.ubuntu.com/ubuntu jammy-updates/main amd64 Packages [602 kB]
#6 130.3 Get:17 http://archive.ubuntu.com/ubuntu jammy-backports/universe amd64 Packages [5814 B]

#6 132.3 Reading state information...
#6 132.3 E: Unable to locate package apt-get
#6 132.3 E: Unable to locate package autoclean
```

I'm hoping this issue is something on github action's side and will resolve itself.

**Comment by erinyoung** [comment]
No worries! It was a simple typo.

**Comment by kapsakcj** [comment]
Just FYI - Looks like they've released a few different versions recently, latest is 13.35.0 

https://github.com/ncbi/datasets/releases

Not sure if there are any bugfixes and/or new features that would warrant updating the version

**Comment by kapsakcj** [comment]
I've made blank repos on dockerhub and quay in preparation for this new docker image

https://hub.docker.com/r/staphb/ncbi-datasets
https://quay.io/repository/staphb/ncbi-datasets?tab=tags

File: ncbi-datasets/13.31.0/README.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# NCBI datasets and dataformat container
+
+Main tool : [datasets](https://www.ncbi.nlm.nih.gov/datasets/docs/v1/download-and-install/#use-the-datasets-tool-to-download-biological-data) and [dataformat](https://www.ncbi.nlm.nih.gov/datasets/docs/v1/download-and-install/#use-the-dataformat-tool-to-convert-data-reports-to-other-formats)
+
+Full documentation: [https://www.ncbi.nlm.nih.gov/datasets/docs/v1/how-tos/](https://www.ncbi.nlm.nih.gov/datasets/docs/v1/how-tos/)
+
+| Use NCBI Datasets to gather metadata, download data packages, view reports and more
```
Comment (kapsakcj) [inline ncbi-datasets/13.31.0/README.md:7]: accidental pipe here? If intending to quote, use a `>` prior to the text

> like so

**Comment by kapsakcj** [comment]
Built successfully locally and tested by downloading a Salmonella Infantis genome:

```bash
$ docker build -t erin/ncbi-datasets:13.31.0 ncbi-datasets/13.31.0/
Sending build context to Docker daemon  4.608kB
Step 1/19 : FROM ubuntu:jammy as app
 ---> d2e4e1f51132
Step 2/19 : ARG DATASETS_VER="13.31.0"
 ---> Running in 993671f8aee0
[...]
Step 19/19 : RUN datasets download virus genome accession ON924087.1 --filename ON924087.1.zip &&   unzip ON924087.1.zip &&   rm ON924087.1.zip &&   cp ncbi_dataset/data/genomic.fna ON924087.1.fna &&   wc -c ON924087.1.fna
 ---> Running in b3ba59950b7a
Downloading: ON924087.1.zip    50.3kB done
New version of client (13.35.0) available at https://ftp.ncbi.nlm.nih.gov/pub/datasets/command-line/LATEST/linux-amd64/datasets
Archive:  ON924087.1.zip
  inflating: README.md               
  inflating: ncbi_dataset/data/data_report.jsonl  
  inflating: ncbi_dataset/data/cds.fna  
  inflating: ncbi_dataset/data/genomic.fna  
  inflating: ncbi_dataset/data/protein.faa  
  inflating: ncbi_dataset/data/virus_dataset.md  
  inflating: ncbi_dataset/data/dataset_catalog.json  
30297 ON924087.1.fna
Removing intermediate container b3ba59950b7a
 ---> ae3bc6b77572
Successfully built ae3bc6b77572
Successfully tagged erin/ncbi-datasets:13.31.0

# enter into interactive shell
$ docker_run -ti erin/ncbi-datasets:13.31.0

I have no name!@f15adb88f1ce:/data$ datasets download genome accession GCA_002592235.1 --filename GCA_002592235.1.zip
Downloading: GCA_002592235.1.zip    4.13MB done
New version of client (13.35.0) available at https://ftp.ncbi.nlm.nih.gov/pub/datasets/command-line/LATEST/linux-amd64/datasets

# unzip the zip archive
I have no name!@f15adb88f1ce:/data$ unzip GCA_002592235.1.zip 
Archive:  GCA_002592235.1.zip
  inflating: README.md               
  inflating: ncbi_dataset/data/assembly_data_report.jsonl  
  inflating: ncbi_dataset/data/GCA_002592235.1/GCA_002592235.1_ASM259223v1_genomic.fna  
  inflating: ncbi_dataset/data/GCA_002592235.1/genomic.gff  
  inflating: ncbi_dataset/data/GCA_002592235.1/protein.faa  
  inflating: ncbi_dataset/data/GCA_002592235.1/cds_from_genomic.fna  
  inflating: ncbi_dataset/data/GCA_002592235.1/sequence_report.jsonl  
  inflating: ncbi_dataset/data/dataset_catalog.json  

# show first line of FASTA file
I have no name!@f15adb88f1ce:/data$ head -n 1 ncbi_dataset/data/GCA_002592235.1/GCA_002592235.1_ASM259223v1_genomic.fna
>JABK01000001.1 Salmonella enterica subsp. enterica serovar Infantis str. CVM 17958 17958_R1_trimmed_contig_1, whole genome shotgun sequence

# show file sizes
$ ls -lh ncbi_dataset/data/GCA_002592235.1/
total 14M
-rw------- 1 1001 1002 4.6M Aug 23 15:11 GCA_002592235.1_ASM259223v1_genomic.fna
-rw------- 1 1001 1002 4.8M Aug 23 15:11 cds_from_genomic.fna
-rw------- 1 1001 1002 2.3M Aug 23 15:11 genomic.gff
-rw------- 1 1001 1002 1.8M Aug 23 15:11 protein.faa
-rw------- 1 1001 1002  13K Aug 23 15:11 sequence_report.jsonl
```

**Review by kapsakcj — APPROVED** [review]
I approve this PR. Just have a few small comments for improvements.

You can upgrade to the latest version of ncbi-datasets if you'd like, but I'm fine to leave as is too. Up to you.

File: README.md [hunk 2]
```diff
@@ -102,6 +102,7 @@ To learn more about the docker pull rate limits and the open source software pro
...
 | [ngmaster](https://hub.docker.com/r/staphb/ngmaster) <br/> [![docker pulls](https://badgen.net/docker/pulls/staphb/ngmaster)](https://hub.docker.com/r/staphb/ngmaster) | <ul><li>0.5.8</li></ul> | https://github.com/MDU-PHL/ngmaster |
 | [NCBI AMRFinderPlus](https://hub.docker.com/r/staphb/ncbi-amrfinderplus) <br/> [![docker pulls](https://badgen.net/docker/pulls/staphb/ncbi-amrfinderplus)](https://hub.docker.com/r/staphb/ncbi-amrfinderplus) | <ul><li>3.1.1b</li><li>3.8.4</li><li>3.8.28</li><li>3.9.3</li><li>3.9.8</li><li>3.10.1</li><li>3.10.5</li><li>3.10.16</li><li>3.10.20</li><li>3.10.24</li><li>3.10.30</li></ul> | [https://github.com/ncbi/amr](https://github.com/ncbi/amr) |
+| [NCBI Datasets](https://hub.docker.com/r/staphb/ncbi-datasets) <br/> [![docker pulls](https://badgen.net/docker/pulls/staphb/ncbi-datasets)](https://hub.docker.com/r/staphb/ncbi-datasets) | <ul><li>13.31.0</li></ul> | [https://github.com/ncbi/datasets](https://github.com/ncbi/datasets) |
```
[the remark this answers is not in the document]
Comment (erinyoung) [reply README.md]: Will do!

File: ncbi-datasets/13.31.0/README.md (hunk 1 above)
Comment (erinyoung) [reply ncbi-datasets/13.31.0/README.md]: definitely an accident. Sorry!

**Comment by erinyoung** [comment]
I just created a file for the newest version as well, and then updated the main readme. This repo seems to update A LOT.

**Comment by kapsakcj** [comment]
Looks good! Tests pass, so I'll merge and deploy these docker images

**Pull request merged**